File: web-interface/tpl/www/bloc/service/ipbx/asterisk/dialaction/queueskill.php
<?php

#
# XiVO Web-Interface
# Copyright (C) 2006-2010  Proformatique <technique@proformatique.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

$form 	 = &$this->get_module('form');
$url 	 = &$this->get_module('url');
$dhtml 	 = &$this->get_module('dhtml');

$element = $this->get_var('element');
$event 	 = $this->get_var('event');
$skills_tree = $this->get_var('skills_tree');

echo '<div id="fd-dialaction-',$event,'-queueskill-actiontype" class="b-nodisplay">',
     $form->select(array('desc'	=> $this->bbf('fm_dialaction_queueskill-skill'),
			    'name'	=> 'dialaction['.$event.'][skill]',
			    'labelid'	=> 'dialaction-'.$event.'-queueskill-skill',
			    'key'	=> 'name',
			    'altkey'	=> 'name',
			    'empty'     => true,
			    'optgroup'	=> array(
				'key'		=> 'category', 
				'unique' 	=> true,
			    )),
		      $skills_tree,
		     'onchange="xivo_ast_defapplication_queueskill_onskillchange(\''.$dhtml->escape($event).'\',\'it-voicemenu-flow\');"'),

     $form->text(array('desc'	=> $this->bbf('fm_dialaction_queueskill-varname'),
			  'name'	=> 'dialaction['.$event.'][varname]',
			  'labelid'	=> 'dialaction-'.$event.'-queueskill-varname',
			  'size'	=> 15,
			  'value'	=> $this->get_var('dialaction',$event,'queueskill','varname'))),

     $form->button(array('name'		=> 'add-defapplication-queueskill',
			 'id'		=> 'it-add-defapplication-queueskill',
			 'value'	=> $this->bbf('fm_bt-add')),
 		         'onclick="xivo_ast_defapplication_queueskill(\''.$dhtml->escape($event).'\',\'it-voicemenu-flow\');"'),

     '</div>';

?>

File: web-interface/unittest/xivojson/queueskill.py
# -*- coding: utf-8 -*-
from __future__ import with_statement

__version__ = "$Revision$ $Date$"
__author__  = "Guillaume Bour <gbour@proformatique.com>"
__license__ = """
    Copyright (C) 2010  Proformatique

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA..
"""
import unittest, pprint, cjson
from xivojson import *
# define global variables
from globals  import *


class Test20QueueSkill(unittest.TestCase):
    def setUp(self):
        global IP, PORT, SSL, USERNAME, PASSWORD

        self.client = JSONClient(IP, PORT, SSL, USERNAME, PASSWORD)
        self.obj    = 'queueskills'

    def test_01_queueskill(self):
        (resp, data) = self.client.list(self.obj)
        self.assertEqual(resp.status, 200)
        
        data = cjson.decode(data)
###        pprint.pprint(data)
        count = len(data)
        
        # ADD
        with open('xivojson/queueskill.json') as f:
            content = cjson.decode(f.read())
            
#        print content
        (resp, data) = self.client.add(self.obj, content)
#        pprint.pprint(data)
        self.assertEqual(resp.status, 200)
        # data == queueskill ID
        id = cjson.decode(data)

#        # LIST / Check add
        (resp, data) = self.client.list(self.obj)
        self.assertEqual(resp.status, 200)
#        
        data = cjson.decode(data)
#        pprint.pprint(data)
        count2 = len(data)
        self.assertEqual(count2, count+1)

#        # SEARCH
        (resp, data) = self.client.view(self.obj, id)
        self.assertEqual(resp.status, 200)

        data = cjson.decode(data)
#        pprint.pprint(data)
        self.assertTrue('name' in data)
        self.assertTrue(data['name'] == 'saucisse')

#        # DELETE
        (resp, data) = self.client.delete(self.obj, id)
        self.assertEqual(resp.status, 200)
#        pprint.pprint(data)
        
#        # try to redelete => must return 404
        (resp, data) = self.client.delete(self.obj, id)
        self.assertEqual(resp.status, 404)


